corehq/change_ownership.py

```diff
@@ -0,0 +1,202 @@
+    user_count = len(users)
+    success_count = fail_count = skip_count = 0
+    total_time = 0
+    print(f"Total Users to Process: {user_count}")
+    user_to_save = []
+    for idx, user in enumerate(users):
+        start_time = time.time()
+        percentage_done = round((idx / user_count) * 100, 2)
+        print(f'Processing User {idx}/{user_count} ({percentage_done}%)')
+
+        # First make sure that the user type is rc
+        user_data = user.get_user_data(DOMAIN)
+        if user_data[USER_TYPE_PROP_NAME] != 'rc':
+            write_to_log([user.user_id], is_success=False, message="User Type not RC")
+            skip_count += 1
+            continue
+
+        try:
+            # Get a descendant of user location which has the same rc number
+            loc = SQLLocation.objects.get(
+                domain=DOMAIN,
+                parent__location_id=user.location_id,
+                name=user_data[RC_NUM_PROP_NAME]
+            )
+            # Set this new descendant location as user location
+            user.location_id = loc.location_id
+            user_to_save.append(user)
+            write_to_log([user.user_id])
+            success_count += 1
+        except SQLLocation.DoesNotExist:
+            write_to_log(
+                [user.user_id],
+                is_success=False,
+                message=f'({user_data[RC_NUM_PROP_NAME]}) does not exist as child of location with id ({loc.location_id})'
+            )
+            fail_count += 1
+        finally:
+            # Time logging
+            end_time = time.time()
+            time_diff = end_time - start_time
+            total_time += time_diff
+            print(f'Time to process User #{idx}: {time_diff}s')
+            print(f'Estimate time remaining: {time_diff * (user_count - idx)}s')
+
+        print("Processing Users Complete!")
+        print(
+            f"Success: {success_count}, 
+            Fail: {fail_count}, 
+            Skipped: {skip_count},
+            Total Time: {round(total_time / 60, 2)} minutes"
+        )
+    CommCareUser.bulk_save(user_to_save)
+    print("Saving Users Complete!")
+
+### Task 2 & 3
+def transfer_case_ownership():
+    print("---MOVING CASE OWNERSHIP---")
+    case_ids = (
+        CaseSearchES()
+        .domain(DOMAIN)
+        .OR(
+            case_type('menage'),
+            case_type('membre'),
+            case_type('seance_educative'),
+            case_type('fiche_pointage')
+        )
+        .sort('opened_on')  # sort so that we can continue
+    ).get_ids()
+
+    case_count = len(case_ids)
+    batch_count = math.ceil(case_count / BATCH_SIZE)
+    print(f'Total Cases to Process: {case_count}')
+    print(f'Total Batches to Process: {batch_count}')
+    case_blocks = []
+    current_chunk = 0
+    end_time = total_time = 0
+    success_count = fail_count = 0
+    start_time = time.time()
+    for case_obj in CommCareCase.objects.iter_cases(case_ids, domain=DOMAIN):
```

Comment:
You should consider using `chunked` here along with `with_progress_bar` like [here](https://github.com/dimagi/commcare-hq/blob/e4713e45c1f1e4c66eecf337c2ceb81af8f53729/corehq/apps/users/management/commands/add_data_dict_permissions.py#L22).
This way you can see progress and have better control on the chunk size which is 100 by default, but you can do more I guess, though even 100 is okay

Reply:
I've implemented chunking along with the `with_progress_bar` function for both users and cases ([bf3b264](https://github.com/dimagi/commcare-hq/pull/34588/commits/bf3b264f740a226213cd2df2fbe86f6fd632fcb8))